### PR TITLE
Feature/161 subsecond decimals

### PR DESCRIPTION
**Pull request by EEParker**
The first commit adds a sub-second decimal option to address #161. 

The second commit adds a flag for disabling RenderedMessage per #167.

These both seem optional, but could be nice to have. I've included them here for review.

### CODE REVIEW

**Comment by EEParker** [comment]
@VictorioBerra If you have time I would like your opinion/review on these changes. Thank you

**Comment by VictorioBerra** [comment]
Looks like the link to the docs is dead: http://dev.splunk.com/view/SP-CAAAE6P

Is there a more up to date link describing the tiemstamp format and the precision?

**Comment by EEParker** [comment]
https://docs.splunk.com/Documentation/Splunk/9.2.0/SearchReference/Commontimeformatvariables it looks like 3, 6 or 9 decimals are supported, so the PR probably needs to be updated to use an enum for millisecond, microseconds or nanoseconds.

**Comment by VictorioBerra** [comment]
That is very helpful! Thank you.

**Comment by EEParker** [comment]
I've updated this PR with the enum, added unit tests and updated documentation links.

**Comment by EEParker** [comment]
I've updated this PR with the enum, added unit tests and updated documentation links.

File: src/Serilog.Sinks.Splunk/Sinks/Splunk/SplunkJsonFormatter.cs [hunk 1]
```diff
@@ -174,8 +182,11 @@ public void Format(LogEvent logEvent, TextWriter output)
                 JsonValueFormatter.WriteQuotedJsonString(logEvent.MessageTemplate.Text, output);
             }
 
-            output.Write(",\"RenderedMessage\":");
-            JsonValueFormatter.WriteQuotedJsonString(logEvent.RenderMessage(_formatProvider), output);
+            if (!_renderMessage)
```
Comment (MatthewHays) [inline src/Serilog.Sinks.Splunk/Sinks/Splunk/SplunkJsonFormatter.cs:185]: Should this logic be the other way round? ie `if (_renderMessage)` to match the _renderTemplate logic above?

**Review by MatthewHays — COMMENTED** [review]
Small comment as to whether _renderMessage=true means to enable or disable the field from being sent to splunk, to match the meaning for _renderTemplate=true

File: src/Serilog.Sinks.Splunk/Sinks/Splunk/SplunkJsonFormatter.cs (hunk 1 above)
Comment (VictorioBerra) [reply src/Serilog.Sinks.Splunk/Sinks/Splunk/SplunkJsonFormatter.cs]: I think you are right. I removed the `!`

Comment (EEParker) [reply src/Serilog.Sinks.Splunk/Sinks/Splunk/SplunkJsonFormatter.cs]: thanks for the fix

**Comment by EEParker** [comment]
I added the test code and splunk props.conf to make this work in the sample project.

**Comment by VictorioBerra** [comment]
Thanks for doing that, @EEParker can you share the changes you made to the props.conf? Maybe I could post on the Splunk forums about this?

**Comment by EEParker** [comment]
> Thanks for doing that, @EEParker can you share the changes you made to the props.conf? Maybe I could post on the Splunk forums about this?

props.conf
```ini
TIME_PREFIX = \"time\"\:\s*\"
TIME_FORMAT = %Y-%m-%dT%H:%M:%S.%9Q%Z
```

https://github.com/serilog-contrib/serilog-sinks-splunk/pull/172/files#diff-7bdfdc66d9929e53da33186cc68b1ccf09bb584d6b50ce4c0be241b259420395R1-R2

**Comment by EEParker** [comment]
Are you comfortable with merging this? I think a follow-up PR can be made to address any splunk samples that need to be changed. I would like to merge this in before #174.

**Comment by VictorioBerra** [comment]
@EEParker Yeah id say go ahead, I like the addition of the .editoconfig and the docker additions.

**Pull request merged**